back/src/Guide/Types/Edit.hs

```diff
@@ -166,15 +166,15 @@ genVer ''Edit 7 [
   Copy 'Edit'MoveItem,
   Copy 'Edit'MoveTrait ]
 
-deriveSafeCopySimple 7 'base ''Edit_v7
+deriveSafeCopySimple 8 'base ''Edit_v8
 
 instance Migrate Edit where
-  type MigrateFrom Edit = Edit_v7
-  migrate = $(migrateVer ''Edit 7 [
+  type MigrateFrom Edit = Edit_v8
+  migrate = $(migrateVer ''Edit 8 [
     CustomM "Edit'AddCategory" [|\x ->
         Edit'AddCategory
-          { editCategoryUid = editCategoryUid_v7 x
-          , editCategoryTitle = editCategoryTitle_v7 x
```

Comment:
This should remain in the old migration; basically, once a migration is written it shouldn't be changed. Only the Hackage stuff should be in the new migration.

Reply:
Your new migration should look something like this:

```haskell
genVer ''Edit 8 [
  -- Add
  Copy 'Edit'AddCategory,
  -- ...all other constructors from current Edit should also be present,
  -- except for Edit'SetItemKind and Edit'SetItemHackage; we want to say 
  -- that in ver 8 we had all the same constructors, but we can't take the
  -- definition of Edit'SetItemKind from the current scope because it
  -- doesn't exist anymore. So in order to recreate the previous version
  -- (i.e. the one that existed in code before your PR) you have to explicitly
  -- tell the migration that this constructor existed and had such-and-such
  -- fields.
  Custom "Edit'SetItemKind" [
      ...field names and types...
      ]

instance Migrate Edit where
  type MigrateFrom Edit = Edit_v8
  migrate = $(migrateVer ''Edit 8 [
    -- Same; all constructors should be copied
    CustomM "Edit'SetItemKind" [|\x ->
      -- here you need to convert Edit'SetItemKind into Edit'SetItemHackage
    |]
```